docs/articles/apex/refactoring.md

```diff
@@ -15,3 +15,25 @@ Validation of the new symbol name is performed before applying the rename. If th
 If the new name conflicts with an existing identifier name, we try to fully qualify the references to the existing name in contexts where the conflicts exist.
 
 ![GIF showing a renaming conflict](../../images/apex-rename-conflict.gif)
+
+## Refactor: Extract Constant
+
+You can extract valid literal expressions into constants. Literal expressions include: String, Integer, Long, Double, Decimal, and Boolean.
+
+This constant will be instantiated below the class declaration. If the expression is contained within an innerclass, the constant will be instantiated below the innerclass declaration. The new constant will always be defined with modifiers private, static, and final.
+
+At this time, there is no conflict detection for the new constant.
+
+![GIF showing extract constant](../../images/extract-constant.gif)
+
+## Refactor: Extract Local Variable
+
+You can extract valid expression types into local variables. Please note that not all expression types are currently supported.
```

Comment:
"expression types" is an implementation detail just "expression" should be fine. Also, it is not clear what "valid" means we could just remove it as well. We don't have to specify that we don't support some expression types since they are just bug fixes which will be handled later.

In general, this is not a specification so there is no need to explicitly go into a lot of details about these features. Our documentation should just be brief enough to describe the feature and how to use it. For ideas you could refer to:
https://www.jetbrains.com/help/idea/extract-variable.html